src/functions.php

```diff
@@ -77,7 +77,7 @@ function uri_for($uri)
 function stream_for($resource = '', array $options = [])
 {
     switch (gettype($resource)) {
-        case 'string':
+        case is_scalar($resource):
```

Comment:
That's true. If we're ok with just documenting it as "Scalar values will be cast to a string using (string), and object will be cast to a string using __toString()", then I think it should be fine... What do you think?